packages/core/src/codewhisperer/service/transformByQ/transformApiHandler.ts

```diff
@@ -385,12 +385,135 @@ export function getImageAsBase64(filePath: string) {
     return `data:image/svg+xml;base64,${fileContents}`
 }
 
+/*
+ * Given the icon name from core/resources/icons/aws/amazonq, get the appropriate icon according to the user's theme.
+ * ex. getIcon('transform-file') returns the 'transform-file-light.svg' icon if user has a light theme enabled,
+ * otherwise 'transform-file-dark.svg' is returned.
+ */
+export function getTransformationIcon(name: string) {
+    let iconPath = ''
+    switch (name) {
+        case 'linesOfCode':
+            iconPath = 'transform-variables'
+            break
+        case 'plannedDependencyChanges':
+            iconPath = 'transform-dependencies'
+            break
+        case 'plannedDeprecatedApiChanges':
+            iconPath = 'transform-step-into'
+            break
+        case 'plannedFileChanges':
+            iconPath = 'transform-file'
+            break
+        case 'upArrow':
+            iconPath = 'transform-arrow'
+            break
+        case 'transformLogo':
+            return getImageAsBase64(globals.context.asAbsolutePath('resources/icons/aws/amazonq/transform-logo.svg'))
+        default:
+            iconPath = 'transform-default'
+            break
+    }
+    const themeColor = vscode.window.activeColorTheme.kind
+    if (themeColor === vscode.ColorThemeKind.Light || themeColor === vscode.ColorThemeKind.HighContrastLight) {
+        iconPath += '-light.svg'
+    } else {
+        iconPath += '-dark.svg'
+    }
+    return getImageAsBase64(globals.context.asAbsolutePath(path.join('resources/icons/aws/amazonq', iconPath)))
+}
+
+export function getFormattedString(s: string) {
+    switch (s) {
+        case 'linesOfCode':
+            return CodeWhispererConstants.linesOfCodeMessage
+        case 'plannedDependencyChanges':
+            return CodeWhispererConstants.dependenciesToBeReplacedMessage
+        case 'plannedDeprecatedApiChanges':
+            return CodeWhispererConstants.deprecatedCodeToBeReplacedMessage
+        case 'plannedFileChanges':
+            return CodeWhispererConstants.filesToBeChangedMessage
+        case 'dependencyName':
```

Comment:
not a blocker (future followup): can this be modeled as a map? instead of a bunch of loose constants mapped by a switch.